Sources/Core/Microsoft.StreamProcessing/StreamableAPI/StreamableExtensions.cs

```diff
@@ -969,6 +969,43 @@ public static partial class Streamable
             });
         }
 
+        /// <summary>
+        /// Macro to perform a left-outer-join operation.
+        /// </summary>
+        /// <typeparam name="TKey">Type of (mapping) key in the stream</typeparam>
+        /// <typeparam name="TLeft">Type of left input payload in the stream</typeparam>
+        /// <typeparam name="TRight">Type of right input payload in the stream</typeparam>
+        /// <typeparam name="TJoinKey">Type of join key for the join</typeparam>
+        /// <typeparam name="TResult">Type of result payload in the stream</typeparam>
+        /// <param name="left">Left input stream</param>
+        /// <param name="right">Right input stream</param>
+        /// <param name="leftKeySelector">Selector for the left-side join key</param>
+        /// <param name="rightKeySelector">Selector for the right-side join key</param>
+        /// <param name="leftResultSelector">Selector for the result for non-joining tuples</param>
+        /// <param name="rightResultSelector">Selector for the result for non-joining tuples</param>
```

Comment:
Probably should add "from the left-side join input" and "from the right-side join input" to these documentation comments for left(right)ResultSelector.